### ISSUE TITLE
supporting using more efficient format for put/delete requests

### ISSUE THREAD

**Issue by BusyJay** [tikv/tikv#12020]
### Is your feature request related to a problem? Please describe:
When using single node, loading a lot of key values shows that about 30% of raftstore CPU is used in propose_normal. They are freeing memory and encoding RaftCmdRequest. This makes the commit duration not match the append duration even for single node.
![image](https://user-images.githubusercontent.com/1701473/155519583-cf9d53f6-314c-431c-afc9-479edbae775e.png)

When using protobuf message, every key and value is stored in a separate byte string and every operation is stored in a request. So it will waste a lot of CPU in loops and freeing small byte slice.

### Describe the feature you'd like:
Instead of just using protobuf message, we can store all key/value pairs in one big slice. So when proposing the message, only one byte slice needs to be freed and copied.

A simple example can be adding a new sub messages in `RaftCmdRequest`.
```
message BlobRequest {
    bytes  mutations = 1;
}

message RaftCmdRequest {
    ...
    BlobRequest blob_request = 5;
}
```

@coocood has done a prototype with his own codec that translate into prewrite/commit format. I also done a prototype that using a simple format as following.

Only put and delete need to be encoded in `BlobRequest`, all other operations still use the old format. So it doesn't require change a lot of codes. And the format of `mutations` can be:
```
('p'|'d') key_len key_content (value_len value_content)
```

Merging two requests is as simple as copying byte slice.

My prototype shows that for single node, using new codecs can make the cost of propose_normal ellageiable. And the QPS is increased from 40k to 48k.
![image](https://user-images.githubusercontent.com/1701473/155519657-42d5dec3-26b3-403d-bd6d-94847badd143.png)

A more aggressive version is store the rocksdb writebatch directly in mutations, this can remove further one copy when translating mutations to write batch in apply threads.

### Describe alternatives you've considered:
No.

**Comment by youjiali1995** [comment]
> A more aggressive version is store the rocksdb writebatch directly in mutations, this can remove further one copy when translating mutations to write batch in apply threads.

I remembered @Little-Wallace did it in 2020 and it brings much improvement.

**Comment by BusyJay** [comment]
Yes, see https://github.com/tikv/tikv/issues/7994. This issue is less aggressive and need less changes.

**Comment by hnes** [comment]
@tonyxuqqi  OK, I will try to implement this feature.

**Comment by BusyJay** [comment]
You may also discuss it for potential compatibility. /cc @pingyu

**Comment by hnes** [comment]
@BusyJay OK, Thanks a lot :-D
